test/test_main.py

```diff
@@ -43,6 +50,15 @@ class SymlinkTargetError(Exception):
     pass
 
 
+class SimulatedEventSpec:
+    def __init__(self, event_type, event_name, relation_id=None, remote_app=None, remote_unit=None):
+        self.event_type = event_type
+        self.event_name = event_name
+        self.relation_id = str(relation_id)
```

Comment:
Doesn't seem appropriate to stringify it here. It's assuming a specific use case which should be handled when it becomes relevant.

Reply:
I added it here to cover the following error:
```
            env.update({
                'JUJU_RELATION': rel_name,
                'JUJU_RELATION_ID': event_spec.relation_id,
            })
```
```
Traceback (most recent call last):
  File "/home/dima/src/canonical/juju/operator/test/test_main.py", line 220, in test_multiple_events_handled
    state = self._simulate_event(event_spec, charm_config)
  File "/home/dima/src/canonical/juju/operator/test/test_main.py", line 151, in _simulate_event
    subprocess.check_call([sys.executable, event_hook], env=env)
  File "/usr/lib/python3.7/subprocess.py", line 358, in check_call
    retcode = call(*popenargs, **kwargs)
  File "/usr/lib/python3.7/subprocess.py", line 339, in call
    with Popen(*popenargs, **kwargs) as p:
  File "/usr/lib/python3.7/subprocess.py", line 800, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.7/subprocess.py", line 1462, in _execute_child
    env_list.append(k + b'=' + os.fsencode(v))
  File "/usr/lib/python3.7/os.py", line 810, in fsencode
    filename = fspath(filename)  # Does type-checking of `filename`.
TypeError: expected str, bytes or os.PathLike object, not int
```

Reply:
Will move it to `_simulate_event` instead to make it closer to the use-case.